src/api/spec/models/workflow_spec.rb

```diff
@@ -321,18 +230,12 @@
   describe '#steps' do
     let(:project) { create(:project, name: 'test-project', maintainer: user) }
     let(:package) { create(:package, name: 'test-package', project: project) }
-    let(:extractor_payload) do
-      {
-        scm: 'github',
-        action: 'opened',
-        event: 'pull_request',
-        pr_number: 1,
-        target_repository_full_name: 'iggy/test-project'
-      }
-    end
+    let!(:workflow_run) { create(:workflow_run, scm_vendor: 'github', hook_event: 'pull_request', hook_action: 'opened', token: token) }
 
     before do
       login user
+      allow(workflow_run).to receive_messages(pr_number: 1)
+      allow(workflow_run).to receive(:target_repository_full_name).and_return('iggy/test-project')
```

Comment:
Why do you need to mock those?

Reply:
I just need to ensure the workflow_run returns the values I need to test other stuff. I can either mock those methods or pass the request payload to the workflow_run and let it extract those values.

After implementing this I realized other specs use the second approach. So I'm going to change this for consistency and use the request payload.

Reply:
Done